Source/Actors/Player.cs

```diff
@@ -1,2364 +1,2377 @@
+		&& stateMachine.State != States.StrawbGet
+		&& stateMachine.State != States.Cassette
+		&& stateMachine.State != States.Dead;
+
+	public Player()
+	{
+		PointShadowAlpha = 1.0f;
+		LocalBounds = new BoundingBox(new Vec3(0, 0, 10), 10);
+		UpdateOffScreen = true;
+
+		// setup model
+		{
+			Model = new(Assets.Models["player"]);
+			Model.SetBlendDuration("Idle", "Dash", 0.05f);
+			Model.SetBlendDuration("Idle", "Run", 0.2f);
+			Model.SetBlendDuration("Run", "Skid", .125f);
+			Model.SetLooping("Dash", false);
+			Model.Flags |= ModelFlags.Silhouette;
+			Model.Play("Idle");
+
+			foreach (var mat in Model.Materials)
+				mat.Effects = 0.60f;
+		}
+
+		stateMachine = new();
+		stateMachine.InitState(States.Normal, StNormalUpdate, StNormalEnter, StNormalExit);
+		stateMachine.InitState(States.Dashing, StDashingUpdate, StDashingEnter, StDashingExit);
+		stateMachine.InitState(States.Skidding, StSkiddingUpdate, StSkiddingEnter, StSkiddingExit);
+		stateMachine.InitState(States.Climbing, StClimbingUpdate, StClimbingEnter, StClimbingExit);
+		stateMachine.InitState(States.StrawbGet, StStrawbGetUpdate, StStrawbGetEnter, StStrawbGetExit, StStrawbGetRoutine);
+		stateMachine.InitState(States.FeatherStart, StFeatherStartUpdate, StFeatherStartEnter, StFeatherStartExit);
+		stateMachine.InitState(States.Feather, StFeatherUpdate, StFeatherEnter, StFeatherExit);
+		stateMachine.InitState(States.Respawn, StRespawnUpdate, StRespawnEnter, StRespawnExit);
+		stateMachine.InitState(States.StrawbReveal, null, StStrawbRevealEnter, StStrawbRevealExit, StStrawbRevealRoutine);
+		stateMachine.InitState(States.Cutscene, StCutsceneUpdate, StCutsceneEnter);
+		stateMachine.InitState(States.Dead, StDeadUpdate, StDeadEnter);
+		stateMachine.InitState(States.Bubble, null, null, StBubbleExit, StBubbleRoutine);
+		stateMachine.InitState(States.Cassette, null, null, StCassetteExit, StCassetteRoutine);
+
+		spikeBlockCheck = (spike) =>
+		{
+			return Vec3.Dot(velocity.Normalized(), spike.Direction) < 0.5f;
+		};
+
+		SetHairColor(0xdb2c00);
+	}
+
+	#region Added / Update
+
+	public override void Added()
+	{
+		if (World.Entry.Reason == World.EntryReasons.Respawned)
+		{
+			cameraTargetForward = storedCameraForward;
+			cameraTargetDistance = storedCameraDistance;
+			stateMachine.State = States.Respawn;
+		}
+		else if (World.Entry.Submap && World.Entry.Reason == World.EntryReasons.Entered)
+		{
+			stateMachine.State = States.StrawbReveal;
+		}
+		else
+		{
+			stateMachine.State = States.Normal;
+		}
+
+		sfxWallSlide = World.Add(new Sound(this, Sfx.sfx_wall_slide));
+		sfxFeather = World.Add(new Sound(this, Sfx.sfx_feather_state_active_loop));
+		sfxBubble = World.Add(new Sound(this, Sfx.sfx_bubble_loop));
+
+		cameraOriginPos = Position;
+		GetCameraTarget(out var orig, out var target, out _);
+		World.Camera.LookAt = target;
+		World.Camera.Position = orig;
+
+		freecamPosition = Position;
+	}
+
+	public void UpdateCamera()
+	{
+		//Part 1
```

Comment:
I don't like those "Part" comments, since they don't give you any information.
Probably make them more descriptive, of what the code is actually doing.